prov/sm2/src/sm2_ep.c

```diff
@@ -257,6 +259,8 @@ static void cleanup_shm_resources(struct sm2_ep *ep)
 			"until SHM file is cleared\n",
 			ep->gid, ep->gid);
 	}
+
+	ofi_genlock_unlock(&ep->util_ep.lock);
```

Comment:
You sleep in this function. If you lock and sleep that is bad. I would either remove the retry/sleep or unlock, relock. (I still don't like that sleep.... if we remove it that would make me happier)

Reply:
I can remove the lock b/c we should only be calling this function after we are destroying the endpoint.